pkg/controller/jobframework/reconciler.go

```diff
@@ -243,6 +243,11 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 
 	// 1. make sure there is only a single existing instance of the workload.
 	// If there's no workload exists and job is unsuspended, we'll stop it immediately.
+	// we don't want to construct the workload for a ray cluster that was created by a RayJob
+	if owner := metav1.GetControllerOf(job.Object()); owner != nil && owner.Kind == "RayJob" && job.GVK().Kind == "RayCluster" {
```

Comment:
I wonder if calling `IsOwnerManagedByKueue` here would generalise it. Otherwise, I'd suggest to add a TODO so we can work on generalising that logic. I think this would also apply to #1568.